cdap-app-fabric/src/main/java/io/cdap/cdap/app/job/SupportBundleJob.java

```diff
@@ -0,0 +1,471 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.job;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import io.cdap.cdap.app.store.Store;
+import io.cdap.cdap.app.task.SupportBundlePipelineInfoTask;
+import io.cdap.cdap.app.task.SupportBundlePipelineRunLogTask;
+import io.cdap.cdap.app.task.SupportBundleRuntimeInfoTask;
+import io.cdap.cdap.app.task.SupportBundleSystemLogTask;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.id.Id;
+import io.cdap.cdap.common.logging.LoggingContext;
+import io.cdap.cdap.internal.app.services.ApplicationLifecycleService;
+import io.cdap.cdap.internal.app.store.RunRecordDetail;
+import io.cdap.cdap.logging.context.LoggingContextHelper;
+import io.cdap.cdap.logging.read.LogReader;
+import io.cdap.cdap.metrics.query.MetricsQueryHelper;
+import io.cdap.cdap.proto.ApplicationDetail;
+import io.cdap.cdap.proto.ApplicationRecord;
+import io.cdap.cdap.proto.ProgramType;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.ProgramRunId;
+import javafx.util.Pair;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+public class SupportBundleJob {
+  private static final Logger LOG = LoggerFactory.getLogger(SupportBundleJob.class);
+  private static final ExecutorService executor = Executors.newFixedThreadPool(5);
+  private static final SupportBundleSystemLogTask supportBundleSystemLogTask =
+      new SupportBundleSystemLogTask();
+  private static final SupportBundlePipelineInfoTask supportBundlePipelineInfoTask =
+      new SupportBundlePipelineInfoTask();
+  private static final SupportBundlePipelineRunLogTask supportBundlePipelineRunLogTask =
+      new SupportBundlePipelineRunLogTask();
+  private static final SupportBundleRuntimeInfoTask supportBundleRuntimeInfoTask =
+      new SupportBundleRuntimeInfoTask();
+  private static List<RunRecordDetail> runRecordDetailList;
+  private static Map<String, Integer> retryServiceMap;
+  private static Queue<String> queue;
+  private static Map<RunRecordDetail, JsonObject> runMetricsMap;
+  private static JsonArray tasks;
+  private final JsonObject statusJson;
```

Comment:
Don't operate JSON objects outside of serialization/deserialization. It breaks all java strong typing